src/main/java/org/terasology/launcher/game/GameService.java

```diff
@@ -0,0 +1,164 @@
+ * <p>
+ * Its {@linkplain #GameService() constructor} requires no arguments. Use {@link #start(Path, BaseLauncherSettings)} to
+ * start the game process; the zero-argument form of {@code start()} will not have enough information.
+ * <p>
+ * The Boolean value of this service is true when it believes the game process has started <em>successfully.</em>
+ * There will be some time between when this service is started and when that value is set. It can be observed on
+ * {@link #valueProperty()} or retrieved as {@link #getValue()}.
+ * <p>
+ * The {@link Worker} interface of this Service provides a proxy to its current {@link RunGameTask}.
+ * Many of the methods the Worker interface defines are not used by this task type. In particular, the
+ * {@link Worker#progressProperty() progress} and {@link Worker#workDoneProperty() workDone} properties have no
+ * information for you that reflect the state of the game process.
+ * <p>
+ * This service will be in {@link State#RUNNING RUNNING} state as long as the game process is live. It enters
+ * {@link State#SUCCEEDED SUCCEEDED} after the game process exits with no error code, or {@link State#FAILED FAILED} if
+ * the game failed to start or it terminates with an error. Whether it succeeded or failed, the service will then
+ * reset to {@link State#READY READY} to be started again.
+ * <p>
+ * This service {@linkplain #cancel() does not support cancellation}.
+ * <ul>
+ *   <li>For details on how the arguments to the process are constructed, see the source for {@link GameStarter}.
+ *   <li>For details on how output from the game process is treated, see {@link RunGameTask}.
+ * </ul>
+ */
+public class GameService extends Service<Boolean> {
+    private static final Logger logger = LoggerFactory.getLogger(GameService.class);
+
+    private Path gamePath;
+    private BaseLauncherSettings settings;
+
+    public GameService() {
+        setExecutor(Executors.newSingleThreadExecutor(
+            new ThreadFactoryBuilder()
+                    .setNameFormat("GameService-%d")
+                    .setDaemon(true)
+                    .setUncaughtExceptionHandler(this::exceptionHandler)
+                    .build()
+        ));
+    }
+
+    /**
+     * Start a new game process with these settings.
+     *
+     * @param gamePath the directory under which we will find libs/Teresology.jar, also used as the process's
+     *     working directory
+     * @param settings supplies other settings relevant to configuring a process
+     */
+    @SuppressWarnings("checkstyle:HiddenField")
+    public void start(Path gamePath, BaseLauncherSettings settings) {
+        this.gamePath = gamePath;
+        this.settings = settings;
+
+        start();
+    }
+
+    /**
+     * Use {@link #start(Path, BaseLauncherSettings)} instead.
+     * <p>
+     * It is an error to call this method before providing the configuration.
+     */
+    @Override
+    public void start() {
+        super.start();
```

Comment:
Consider checking that `gamePath` and `settings` are not null, and throw and exception otherwise. 

Guava's `Preconditions` are your friend 😉

Reply:
a question of how many layers to do that checking at. It's done in `createTask`.

On one hand, identifying and reporting the problem earlier is better.  
On the other hand, I don't think this is very much earlier at all.

think it will add clarity if we add the checks here?